metaflow/datastore/task_datastore.py

```diff
@@ -740,8 +801,19 @@ def artifacts_iter():
                     # '_' as they are used by the Metaflow runtime.
                     delattr(flow, var)
                 yield var, val
+                art = flow._orig_artifacts.get(var)
```

Comment:
change in behavior possible here since we keep things we serialize if they require it. For others, no change. Note that the idea would be that `post_serialize` could hollow out the MetaflowArtifact as well to keep memory consumption low.